aspnetcore/security/authentication/azure-ad-b2c.md

```diff
@@ -0,0 +1,119 @@
+
+## Prerequisites
+
+The following are required for this walkthrough:
+
+* A [Microsoft Azure subscription](https://azure.microsoft.com/free/?ref=microsoft.com&utm_source=microsoft.com&utm_medium=docs&utm_campaign=visualstudio).  
+* [Visual Studio 2017](https://www.visualstudio.com/vs/) (any edition).
+
+## Create the Azure Active Directory B2C tenant
+
+Create an Azure Active Directory B2C tenant [as described in the documentation](https://docs.microsoft.com/azure/active-directory-b2c/active-directory-b2c-get-started).  When prompted, associating the tenant with an Azure subscription is optional for this tutorial.
+
+## Register the application in AAD B2C
+
+In the newly created AAD B2C tenant, register your application using [the steps in the documentation under the *Register a web app* heading](https://docs.microsoft.com/azure/active-directory-b2c/active-directory-b2c-app-registration#register-a-web-app). Stop at the *Create a web app client secret* heading. A client secret isn't required for this tutorial. 
+
+Use the following values:
+
+| Setting      | Value  | Notes                                        |
+| ------------ | ------- | -------------------------------------------------- |
+| **Name** | *\<application name\>* | Enter a **Name** for the application that describes your application to consumers. | 
+| **Include web app / web API** | Yes | |
+| **Allow implicit flow** | Yes | |
+| **Reply URL** | `https://localhost:44300` | Reply URLs are endpoints where Azure AD B2C returns any tokens that your application requests. Visual Studio will provide the Reply URL to use. For now, enter `https://localhost:44300` to complete the form.
+| **App ID URI** | Leave blank | | Not required for this tutorial.
+| **Include native client** | No | |
+
+> [!WARNING]
+> If setting up a non-localhost Reply URL, be aware of the [constraints on what can be added to the Reply URL list](https://docs.microsoft.com/azure/active-directory-b2c/active-directory-b2c-app-registration#choosing-a-web-app-or-api-reply-url). 
+
+After the application registration is created, the list of applications in the tenant is displayed. Select the application that was just registered.  Select the **Copy** icon to the right of the **Application ID** field to copy the Application ID to the clipboard.
+
+Nothing more can be configured in the AAD B2C tenant at this time, but leave the browser window open. There will be more configuration after the ASP.NET Core application is created.
+
+## Create an ASP.NET Core application in Visual Studio 2017
+
+The Visual Studio Web Application template can be configured to use the AAD B2C tenant for authentication.
+
+In Visual Studio:
+
+1. Create a new ASP.NET Core Web Application. 
+2. Select Web Application from the list of templates.
+3. Select the **Change Authentication** button.
+    
+    ![Change Authentication Button](./azure-ad-b2c/_static/changeauth.png)
```

Comment:
alt texts should be a bit more detailed to describe to someone with accessibility what they can't see.

examples: https://worldready.cloudapp.net/Styleguide/Read?id=2665&topicid=28357